server/src/main/java/io/crate/execution/ddl/tables/TransportDropPartitionsAction.java

```diff
@@ -0,0 +1,90 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.execution.ddl.tables;
+
+import java.util.List;
+
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.ClusterStateTaskExecutor;
+import org.elasticsearch.cluster.block.ClusterBlockException;
+import org.elasticsearch.cluster.block.ClusterBlockLevel;
+import org.elasticsearch.cluster.metadata.MetadataDeleteIndexService;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Singleton;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import io.crate.execution.ddl.AbstractDDLTransportAction;
+import io.crate.metadata.cluster.DropPartitionsClusterStateTaskExecutor;
+
+@Singleton
+public class TransportDropPartitionsAction extends AbstractDDLTransportAction<DropPartitionsRequest, AcknowledgedResponse> {
+
+    public static final Action ACTION = new Action();
+
+    public static class Action extends ActionType<AcknowledgedResponse> {
+        public static final String NAME = "internal:crate:sql/table/drop-partitions";
+
+        private Action() {
+            super(NAME);
+        }
+    }
+
+    private final DropPartitionsClusterStateTaskExecutor executor;
+
+    @Inject
+    public TransportDropPartitionsAction(TransportService transportService,
+                                         ClusterService clusterService,
+                                         ThreadPool threadPool,
+                                         MetadataDeleteIndexService deleteIndexService) {
+        super(
+            ACTION.name(),
+            transportService,
+            clusterService,
+            threadPool,
+            DropPartitionsRequest::new,
+            AcknowledgedResponse::new,
+            AcknowledgedResponse::new,
+            "drop-table-partitions"
+        );
+        executor = new DropPartitionsClusterStateTaskExecutor(deleteIndexService);
+    }
+
+    @Override
+    public ClusterStateTaskExecutor<DropPartitionsRequest> clusterStateTaskExecutor(DropPartitionsRequest request) {
+        return executor;
+    }
+
+    @Override
+    protected ClusterBlockException checkBlock(DropPartitionsRequest request, ClusterState state) {
+        String[] indexNames = state.metadata().getIndices(
+            request.relationName(),
+            List.of(),
```

Comment:
Shouldn't the partition values be passed in here? Otherwise the list will be empty for partitioned tables...

Reply:
yes, nice catch.